src/test/scala/io/iohk/ethereum/jsonrpc/QaJRCSpec.scala

```diff
@@ -125,9 +124,173 @@ class QaJRCSpec extends AnyWordSpec with Matchers with ScalaFutures with NormalP
         response should haveResult(result)
       }
     }
+
+    "request generating checkpoint and return valid response" when {
+      "given block to be checkpointed exists and checkpoint is generated correctly" in new TestSetup {
+        (qaService.generateCheckpoint _)
+          .expects(generateCheckpointReq)
+          .returning(Future.successful(Right(GenerateCheckpointResponse(checkpoint))))
+
+        val response: JsonRpcResponse =
+          jsonRpcController.handleRequest(generateCheckpointRpcRequest).futureValue
+
+        response should haveResult(Extraction.decompose(checkpoint))
+      }
+    }
+
+    "request generating block with checkpoint and return valid response" when {
+      "requested best block to be checkpointed and block with checkpoint is generated correctly" in new TestSetup {
+        val req = generateCheckpointRpcRequest.copy(
+          params = Some(
+            JArray(
+              List(
+                JArray(
+                  privateKeysAsJson
+                )
+              )
+            )
+          )
+        )
+        val expectedServiceReq = generateCheckpointReq.copy(blockHash = None)
+        (qaService.generateCheckpoint _)
+          .expects(expectedServiceReq)
+          .returning(Future.successful(Right(GenerateCheckpointResponse(checkpoint))))
+
+        val response: JsonRpcResponse =
+          jsonRpcController.handleRequest(req).futureValue
+
+        response should haveResult(Extraction.decompose(checkpoint))
+      }
+    }
+
+    "request generating block with checkpoint and return InvalidParams" when {
+      "given block to be checkpointed doesn't exist in blockchain" in new TestSetup {
+        val errorMsg =
+          s"Block to be checkpointed with hash ${ByteStringUtils.hash2string(blockHash)} doesn't exist"
```

Comment:
Minor: should we start adding somewhere this error messages so that we don't have to be repeating them between codebase and tests?

Reply:
This test doesn't make any sense. It sets an expectation and it tests only that expectation. Also, that expectation has nothing to do with the actual implementation. I have removed it.